crates/provider-sdk/src/provider.rs

```diff
@@ -252,9 +252,10 @@ impl ProviderConnection {
                                     current.record("link_name", &tracing::field::display(&inv.target.link_name));
                                     current.record("payload_size", &tracing::field::display(&inv.content_length));
                                     let inv_id = inv.id.clone();
+                                    let inv_operation = inv.operation.clone();
```

Comment:
Is this clone needed? You might be able to change `handle_rpc` to not take ownership of `inv`. I only checked briefly but didn't see a reason we need to move it

Reply:
Actually, do we need to add `operation` to this error event? It looks like it's already on the `current` span above

Reply:
I thought the same thing, and checked and `handle_rpc` definitely does stuff with a mutable `Invocation` -- in particular `dechunk` will do stuff with it.

For the invocation metadata, I followed the existing pattern that was there with `inv_id` -- it's also on the span but doesn't get used in that way. 

I can see about changing both of them!

- [ ] change `inv_id` and `inv_operation` to be pulled from the span

Reply:
Oh and for where `operation` shows up -- it was very hard to read and parse out -- "invocation failed" was easy to find, but the immediate obvious question was "which invocation?". I think it's better to over-communicate here and prevent frustration.

Reply:
The error log contains all the fields on the span, right? So doesn't adding the operation to the event itself mean it shows up twice in the same log?

If you're saying it's simply more readable this way then that's fine, I'm in favor of easier to read errors 👍

Reply:
Ah so I think it is on the span but IIRC it doesn't show up in the **local** logs.

I went back and broke it to collect some logs:

```
  ... more trace level logs ...

  TRACE wasmcloud_host::wasmbus: resolve target
    at crates/host/src/wasmbus/mod.rs:251
    in wasmcloud_host::wasmbus::resolve_target with target: Some(Link(Some("keyvalue"))), links: Some({"keyvalue": WasmCloudEntity { public_key: "VDDB3GKUMKNR66FEIEZ6IWIUGPY4UTEMRODD2FTDCFJCDKX24AIIA6QQ", link_name: "keyvalue", contract_id: "wasmcloud:keyvalue" }}), aliases: {"builtins-component-reactor-preview2": WasmCloudEntity { public_key: "MDIY3WYMGRUBNIHMNPL2RRKEPAZB2W6W4G575ZTTSTIJVSQAQLKIBXBW", link_name: "", contract_id: "" }, "foobar-component-command-preview2": WasmCloudEntity { public_key: "MCWBTC3GVGZUUEYAELF53I5PUGQW243MZFBEYYASDMPQTJLP47CWKE6A", link_name: "", contract_id: "" }, "builtins-compat-reactor-preview2": WasmCloudEntity { public_key: "MAR4257NBNGO7C3A6YOP3PD4GWAWQWKZQKEICLTORYZK7PPYXPXONBBO", link_name: "", contract_id: "" }, "builtins-module-reactor": WasmCloudEntity { public_key: "MALHWDXB3I3C7YJXQJEYZ2FPVQB5CSWDLA7LKB6REPKGZYSPFJMYRYF3", link_name: "", contract_id: "" }}
    in wasmcloud_host::wasmbus::call_operation_with_payload with target: Some(Link(Some("keyvalue")))
    in wasmcloud_host::wasmbus::call_sync with target: Some(Link(Some("keyvalue"))), operation: "wasmcloud:keyvalue/KeyValue.Get"
    in wasmcloud_runtime::actor::component::bus::call_sync with target: Some(TargetEntity::Link(Some("keyvalue"))), operation: "wasmcloud:keyvalue/KeyValue.Get"
    in wasmcloud_runtime::actor::component::call
    in wasmcloud_runtime::actor::component::http::handle
    in wasmcloud_host::wasmbus::handle_invocation with contract_id: "wasmcloud:httpserver", operation: "HttpServer.HandleRequest"
    in wasmcloud_host::wasmbus::handle_call
    in wasmcloud_host::wasmbus::handle_message

2023-10-02T08:32:15.028952Z ERROR wasmcloud_provider_sdk::provider: Invocation failed error=Error when deserializing invocation: Syntax("invalid type: map, expected a string") operation=KeyValue.Get
  TRACE wasmcloud_runtime::actor::module::wasmbus: set host response, res: [130, 165, 118, 97, 108, 117, 101, 163, 98, 97, 114, 166, 101, 120, 105, 115, 116, 115, 195]
    at crates/runtime/src/actor/module/wasmbus.rs:158
    in wasmcloud_runtime::actor::module::wasmbus::set_host_response
    in wasmcloud_runtime::actor::module::wasmbus::host_call with binding_ptr: 1115664, binding_len: 8, namespace_ptr: 1058604, namespace_len: 18, operation_ptr: 1058623, operation_len: 12, payload_ptr: 1115848, payload_len: 4
    in wasmcloud_runtime::actor::module::call
    in wasmcloud_runtime::actor::module::call
    in wasmcloud_runtime::actor::module::handle
    in wasmcloud_host::wasmbus::handle_invocation with contract_id: "wasmcloud:httpserver", operation: "HttpServer.HandleRequest"
    in wasmcloud_host::wasmbus::handle_call
    in wasmcloud_host::wasmbus::handle_message

  TRACE wasmcloud_runtime::actor::module::wasmbus: wrote bytes into guest memory, buf: [130, 165, 118, 97, 108, 117, 101, 163, 98, 97, 114, 166, 101, 120, 105, 115, 116, 115, 195]
    at crates/runtime/src/actor/module/wasmbus.rs:145
    in wasmcloud_runtime::actor::module::wasmbus::write_bytes with data: 1115984
    in wasmcloud_runtime::actor::module::wasmbus::write response into guest memory with res: [130, 165, 118, 97, 108, 117, 101, 163, 98, 97, 114, 166, 101, 120, 105, 115, 116, 115, 195]
    in wasmcloud_runtime::actor::module::wasmbus::host_response with res_ptr: 1115984
    in wasmcloud_runtime::actor::module::call
    in wasmcloud_runtime::actor::module::call
    in wasmcloud_runtime::actor::module::handle
    in wasmcloud_host::wasmbus::handle_invocation with contract_id: "wasmcloud:httpserver", operation: "HttpServer.HandleRequest"
    in wasmcloud_host::wasmbus::handle_call
    in wasmcloud_host::wasmbus::handle_message

thread '  TRACE wasmcloud_runtime::actor::module::wasmbus: read buffer from guest memory, buf: [107, 101, 121, 118, 97, 108, 117, 101]
    at crates/runtime/src/actor/module/wasmbus.rs:115
    in wasmcloud_runtime::actor::module::wasmbus::read_bytes with data: 1115664, len: 8
    in wasmcloud_runtime::actor::module::wasmbus::read_string with data: 1115664, len: 8
    in wasmcloud_runtime::actor::module::wasmbus::host_call with binding_ptr: 1115664, binding_len: 8, namespace_ptr: 1058744, namespace_len: 18, operation_ptr: 1058763, operation_len: 12, payload_ptr: 1116024, payload_len: 4
    in wasmcloud_runtime::actor::module::call
    in wasmcloud_runtime::actor::module::call
    in wasmcloud_runtime::actor::module::handle
    in wasmcloud_host::wasmbus::handle_invocation with contract_id: "wasmcloud:httpserver", operation: "HttpServer.HandleRequest"
    in wasmcloud_host::wasmbus::handle_call
    in wasmcloud_host::wasmbus::handle_message

  TRACE wasmcloud_runtime::actor::module::wasmbus: read string from guest memory, s: "keyvalue"
    at crates/runtime/src/actor/module/wasmbus.rs:129
    in wasmcloud_runtime::actor::module::wasmbus::read_string with data: 1115664, len: 8
    in wasmcloud_runtime::actor::module::wasmbus::host_call with binding_ptr: 1115664, binding_len: 8, namespace_ptr: 1058744, namespace_len: 18, operation_ptr: 1058763, operation_len: 12, payload_ptr: 1116024, payload_len: 4
    in wasmcloud_runtime::actor::module::call
    in wasmcloud_runtime::actor::module::call
    in wasmcloud_runtime::actor::module::handle
    in wasmcloud_host::wasmbus::handle_invocation with contract_id: "wasmcloud:httpserver", operation: "HttpServer.HandleRequest"
    in wasmcloud_host::wasmbus::handle_call
    in wasmcloud_host::wasmbus::handle_message

  TRACE wasmcloud_runtime::actor::module::wasmbus: read buffer from guest memory, buf: [119, 97, 115, 109, 99, 108, 111, 117, 100, 58, 107, 101, 121, 118, 97, 108, 117, 101]
    at crates/runtime/src/actor/module/wasmbus.rs:115
    in wasmcloud_runtime::actor::module::wasmbus::read_bytes with data: 1058744, len: 18
    in wasmcloud_runtime::actor::module::wasmbus::read_string with data: 1058744, len: 18
    in wasmcloud_runtime::actor::module::wasmbus::host_call with binding_ptr: 1115664, binding_len: 8, namespace_ptr: 1058744, namespace_len: 18, operation_ptr: 1058763, operation_len: 12, payload_ptr: 1116024, payload_len: 4
    in wasmcloud_runtime::actor::module::call
    in wasmcloud_runtime::actor::module::call
    in wasmcloud_runtime::actor::module::handle
    in wasmcloud_host::wasmbus::handle_invocation with contract_id: "wasmcloud:httpserver", operation: "HttpServer.HandleRequest"
    in wasmcloud_host::wasmbus::handle_call
    in wasmcloud_host::wasmbus::handle_message

<unnamed>  TRACE wasmcloud_runtime::actor::module::wasmbus: read string from guest memory, s: "wasmcloud:keyvalue"
    at crates/runtime/src/actor/module/wasmbus.rs:129
    in wasmcloud_runtime::actor::module::wasmbus::read_string with data: 1058744, len: 18
    in wasmcloud_runtime::actor::module::wasmbus::host_call with binding_ptr: 1115664, binding_len: 8, namespace_ptr: 1058744, namespace_len: 18, operation_ptr: 1058763, operation_len: 12, payload_ptr: 1116024, payload_len: 4
    in wasmcloud_runtime::actor::module::call
    in wasmcloud_runtime::actor::module::call
    in wasmcloud_runtime::actor::module::handle
    in wasmcloud_host::wasmbus::handle_invocation with contract_id: "wasmcloud:httpserver", operation: "HttpServer.HandleRequest"
    in wasmcloud_host::wasmbus::handle_call
    in wasmcloud_host::wasmbus::handle_message

  TRACE wasmcloud_runtime::actor::module::wasmbus: read buffer from guest memory, buf: [75, 101, 121, 86, 97, 108, 117, 101, 46, 68, 101, 108]
    at crates/runtime/src/actor/module/wasmbus.rs:115
    in wasmcloud_runtime::actor::module::wasmbus::read_bytes with data: 1058763, len: 12
    in wasmcloud_runtime::actor::module::wasmbus::read_string with data: 1058763, len: 12
    in wasmcloud_runtime::actor::module::wasmbus::host_call with binding_ptr: 1115664, binding_len: 8, namespace_ptr: 1058744, namespace_len: 18, operation_ptr: 1058763, operation_len: 12, payload_ptr: 1116024, payload_len: 4
    in wasmcloud_runtime::actor::module::call
    in wasmcloud_runtime::actor::module::call
    in wasmcloud_runtime::actor::module::handle
    in wasmcloud_host::wasmbus::handle_invocation with contract_id: "wasmcloud:httpserver", operation: "HttpServer.HandleRequest"
    in wasmcloud_host::wasmbus::handle_call
    in wasmcloud_host::wasmbus::handle_message

  TRACE wasmcloud_runtime::actor::module::wasmbus: read string from guest memory, s: "KeyValue.Del"
    at crates/runtime/src/actor/module/wasmbus.rs:129
    in wasmcloud_runtime::actor::module::wasmbus::read_string with data: 1058763, len: 12
    in wasmcloud_runtime::actor::module::wasmbus::host_call with binding_ptr: 1115664, binding_len: 8, namespace_ptr: 1058744, namespace_len: 18, operation_ptr: 1058763, operation_len: 12, payload_ptr: 1116024, payload_len: 4
    in wasmcloud_runtime::actor::module::call
    in wasmcloud_runtime::actor::module::call
    in wasmcloud_runtime::actor::module::handle
    in wasmcloud_host::wasmbus::handle_invocation with contract_id: "wasmcloud:httpserver", operation: "HttpServer.HandleRequest"
    in wasmcloud_host::wasmbus::handle_call
    in wasmcloud_host::wasmbus::handle_message

  TRACE wasmcloud_runtime::actor::module::wasmbus: read buffer from guest memory, buf: [163, 102, 111, 111]
    at crates/runtime/src/actor/module/wasmbus.rs:115
    in wasmcloud_runtime::actor::module::wasmbus::read_bytes with data: 1116024, len: 4
    in wasmcloud_runtime::actor::module::wasmbus::host_call with binding_ptr: 1115664, binding_len: 8, namespace_ptr: 1058744, namespace_len: 18, operation_ptr: 1058763, operation_len: 12, payload_ptr: 1116024, payload_len: 4
    in wasmcloud_runtime::actor::module::call
    in wasmcloud_runtime::actor::module::call
    in wasmcloud_runtime::actor::module::handle
    in wasmcloud_host::wasmbus::handle_invocation with contract_id: "wasmcloud:httpserver", operation: "HttpServer.HandleRequest"
    in wasmcloud_host::wasmbus::handle_call
    in wasmcloud_host::wasmbus::handle_message

' panicked at '  TRACE wasmcloud_runtime::capability::builtin: call `Bus` handler
    at crates/runtime/src/capability/builtin.rs:547
    in wasmcloud_runtime::capability::builtin::identify_wasmbus_target with self: Handler { blobstore: "set", bus: "set", incoming_http: "unset", keyvalue_atomic: "set", keyvalue_readwrite: "set", logging: "set", messaging: "set" }, binding: "keyvalue", namespace: "wasmcloud:keyvalue"
    in wasmcloud_runtime::actor::module::wasmbus::handle with binding: "keyvalue", namespace: "wasmcloud:keyvalue", operation: "KeyValue.Del"
    in wasmcloud_runtime::actor::module::wasmbus::host_call with binding_ptr: 1115664, binding_len: 8, namespace_ptr: 1058744, namespace_len: 18, operation_ptr: 1058763, operation_len: 12, payload_ptr: 1116024, payload_len: 4
    in wasmcloud_runtime::actor::module::call
    in wasmcloud_runtime::actor::module::call
    in wasmcloud_runtime::actor::module::handle
    in wasmcloud_host::wasmbus::handle_invocation with contract_id: "wasmcloud:httpserver", operation: "HttpServer.HandleRequest"
    in wasmcloud_host::wasmbus::handle_call
    in wasmcloud_host::wasmbus::handle_message

  TRACE wasmcloud_runtime::capability::builtin: call `Bus` handler
    at crates/runtime/src/capability/builtin.rs:51
    in wasmcloud_runtime::actor::module::wasmbus::handle with binding: "keyvalue", namespace: "wasmcloud:keyvalue", operation: "KeyValue.Del"
    in wasmcloud_runtime::actor::module::wasmbus::host_call with binding_ptr: 1115664, binding_len: 8, namespace_ptr: 1058744, namespace_len: 18, operation_ptr: 1058763, operation_len: 12, payload_ptr: 1116024, payload_len: 4
    in wasmcloud_runtime::actor::module::call
    in wasmcloud_runtime::actor::module::call
    in wasmcloud_runtime::actor::module::handle
    in wasmcloud_host::wasmbus::handle_invocation with contract_id: "wasmcloud:httpserver", operation: "HttpServer.HandleRequest"
    in wasmcloud_host::wasmbus::handle_call
    in wasmcloud_host::wasmbus::handle_message

failed to get `foo`: "call failed: Error when handling invocation: Error when deserializing invocation: Syntax(\"invalid type: map, expected a string\")"  TRACE wasmcloud_host::wasmbus: resolve target
    at crates/host/src/wasmbus/mod.rs:251
    in wasmcloud_host::wasmbus::resolve_target with target: Some(Link(Some("keyvalue"))), links: Some({"keyvalue": WasmCloudEntity { public_key: "VDDB3GKUMKNR66FEIEZ6IWIUGPY4UTEMRODD2FTDCFJCDKX24AIIA6QQ", link_name: "keyvalue", contract_id: "wasmcloud:keyvalue" }}), aliases: {"builtins-component-reactor-preview2": WasmCloudEntity { public_key: "MDIY3WYMGRUBNIHMNPL2RRKEPAZB2W6W4G575ZTTSTIJVSQAQLKIBXBW", link_name: "", contract_id: "" }, "foobar-component-command-preview2": WasmCloudEntity { public_key: "MCWBTC3GVGZUUEYAELF53I5PUGQW243MZFBEYYASDMPQTJLP47CWKE6A", link_name: "", contract_id: "" }, "builtins-compat-reactor-preview2": WasmCloudEntity { public_key: "MAR4257NBNGO7C3A6YOP3PD4GWAWQWKZQKEICLTORYZK7PPYXPXONBBO", link_name: "", contract_id: "" }, "builtins-module-reactor": WasmCloudEntity { public_key: "MALHWDXB3I3C7YJXQJEYZ2FPVQB5CSWDLA7LKB6REPKGZYSPFJMYRYF3", link_name: "", contract_id: "" }}
    in wasmcloud_host::wasmbus::call_operation_with_payload with target: Some(Link(Some("keyvalue")))
    in wasmcloud_host::wasmbus::call_sync with target: Some(Link(Some("keyvalue"))), operation: "wasmcloud:keyvalue/KeyValue.Del"
    in wasmcloud_runtime::actor::module::wasmbus::handle with binding: "keyvalue", namespace: "wasmcloud:keyvalue", operation: "KeyValue.Del"
    in wasmcloud_runtime::actor::module::wasmbus::host_call with binding_ptr: 1115664, binding_len: 8, namespace_ptr: 1058744, namespace_len: 18, operation_ptr: 1058763, operation_len: 12, payload_ptr: 1116024, payload_len: 4
    in wasmcloud_runtime::actor::module::call
    in wasmcloud_runtime::actor::module::call
    in wasmcloud_runtime::actor::module::handle
    in wasmcloud_host::wasmbus::handle_invocation with contract_id: "wasmcloud:httpserver", operation: "HttpServer.HandleRequest"
    in wasmcloud_host::wasmbus::handle_call
    in wasmcloud_host::wasmbus::handle_message

', builtins-component-reactor/src/lib.rs:203:14
note: run with `RUST_BACKTRACE=1` environment variable to display a backtrace
  ERROR wasmcloud_host::wasmbus: failed to handle request, origin: WasmCloudEntity { public_key: "VAUO56KNNCEZAVKXM5K24QE72KAHCQ4RDTWMEOJX5JFG6IPQR3YJ2BXU", link_name: "httpserver", contract_id: "wasmcloud:httpserver" }, target: WasmCloudEntity { public_key: "MDIY3WYMGRUBNIHMNPL2RRKEPAZB2W6W4G575ZTTSTIJVSQAQLKIBXBW", link_name: "", contract_id: "" }, operation: "HttpServer.HandleRequest", invocation_id: "499d59538d674d7987235d86f5d5bf27", e: error while executing at wasm backtrace:
    0: 0x8ad4e - <unknown>!__rust_start_panic
    1: 0x8ab6f - <unknown>!rust_panic
    2: 0x8ab4a - <unknown>!std::panicking::rust_panic_with_hook::hd084a40442e42af7
    3: 0x89eb0 - <unknown>!std::panicking::begin_panic_handler::{{closure}}::h7b63d0dbcdd5afd8
    4: 0x89ddd - <unknown>!std::sys_common::backtrace::__rust_end_short_backtrace::h1c8caee1545e21bb
    5: 0x8a460 - <unknown>!rust_begin_unwind
    6: 0x8fc9e - <unknown>!core::panicking::panic_fmt::hbc45cade774fd748
    7: 0x96577 - <unknown>!core::result::unwrap_failed::h75b87e1db5b0a6ed
    8: 0x378b7 - <unknown>!core::result::Result<T,E>::expect::hfa7870edde143c7d
    9: 0x2f9f4 - <unknown>!<builtins_component_reactor::Actor as builtins_component_reactor::exports::wasi::http::incoming_handler::Guest>::handle::h6101d9e0b4175aee
   10: 0x32c89 - <unknown>!wasi:http/incoming-handler#handle
note: using the `WASMTIME_BACKTRACE_DETAILS=1` environment variable may show more debugging information: wasm trap: wasm `unreachable` instruction executed
    at crates/host/src/wasmbus/mod.rs:1339
    in wasmcloud_host::wasmbus::handle_message

2023-10-02T08:32:15.034485Z  WARN httpserver: actor responded with error error=error while executing at wasm backtrace:
    0: 0x8ad4e - <unknown>!__rust_start_panic
    1: 0x8ab6f - <unknown>!rust_panic
    2: 0x8ab4a - <unknown>!std::panicking::rust_panic_with_hook::hd084a40442e42af7
    3: 0x89eb0 - <unknown>!std::panicking::begin_panic_handler::{{closure}}::h7b63d0dbcdd5afd8
    4: 0x89ddd - <unknown>!std::sys_common::backtrace::__rust_end_short_backtrace::h1c8caee1545e21bb
    5: 0x8a460 - <unknown>!rust_begin_unwind
    6: 0x8fc9e - <unknown>!core::panicking::panic_fmt::hbc45cade774fd748
    7: 0x96577 - <unknown>!core::result::unwrap_failed::h75b87e1db5b0a6ed
    8: 0x378b7 - <unknown>!core::result::Result<T,E>::expect::hfa7870edde143c7d
    9: 0x2f9f4 - <unknown>!<builtins_component_reactor::Actor as builtins_component_reactor::exports::wasi::http::incoming_handler::Guest>::handle::h6101d9e0b4175aee
   10: 0x32c89 - <unknown>!wasi:http/incoming-handler#handle
note: using the `WASMTIME_BACKTRACE_DETAILS=1` environment variable may show more debugging information: wasm trap: wasm `unreachable` instruction executed
2023-10-02T08:32:15.034532Z ERROR wasmcloud_provider_httpserver: Error sending Request to actor error=error while executing at wasm backtrace:
    0: 0x8ad4e - <unknown>!__rust_start_panic
    1: 0x8ab6f - <unknown>!rust_panic
    2: 0x8ab4a - <unknown>!std::panicking::rust_panic_with_hook::hd084a40442e42af7
    3: 0x89eb0 - <unknown>!std::panicking::begin_panic_handler::{{closure}}::h7b63d0dbcdd5afd8
    4: 0x89ddd - <unknown>!std::sys_common::backtrace::__rust_end_short_backtrace::h1c8caee1545e21bb
    5: 0x8a460 - <unknown>!rust_begin_unwind
    6: 0x8fc9e - <unknown>!core::panicking::panic_fmt::hbc45cade774fd748
    7: 0x96577 - <unknown>!core::result::unwrap_failed::h75b87e1db5b0a6ed
    8: 0x378b7 - <unknown>!core::result::Result<T,E>::expect::hfa7870edde143c7d
    9: 0x2f9f4 - <unknown>!<builtins_component_reactor::Actor as builtins_component_reactor::exports::wasi::http::incoming_handler::Guest>::handle::h6101d9e0b4175aee
   10: 0x32c89 - <unknown>!wasi:http/incoming-handler#handle
note: using the `WASMTIME_BACKTRACE_DETAILS=1` environment variable may show more debugging information: wasm trap: wasm `unreachable` instruction executed
2023-10-02T08:32:15.034565Z ERROR warp::filters::trace: unable to process request (internal error) status=500 error=None
  TRACE wasmcloud_runtime::actor::module::wasmbus: set host error, err: "failed to call `wasmcloud:bus/host.call`: failed to call linked provider: failed to publish on NATS topic: request failed: Other"
    at crates/runtime/src/actor/module/wasmbus.rs:151
    in wasmcloud_runtime::actor::module::wasmbus::set_host_error
    in wasmcloud_runtime::actor::module::wasmbus::host_call with binding_ptr: 1115664, binding_len: 8, namespace_ptr: 1058744, namespace_len: 18, operation_ptr: 1058763, operation_len: 12, payload_ptr: 1116024, payload_len: 4
    in wasmcloud_runtime::actor::module::call
    in wasmcloud_runtime::actor::module::call
    in wasmcloud_runtime::actor::module::handle
    in wasmcloud_host::wasmbus::handle_invocation with contract_id: "wasmcloud:httpserver", operation: "HttpServer.HandleRequest"
    in wasmcloud_host::wasmbus::handle_call
    in wasmcloud_host::wasmbus::handle_message

  TRACE wasmcloud_runtime::actor::module::wasmbus: len: 128
    at crates/runtime/src/actor/module/wasmbus.rs:349
    in wasmcloud_runtime::actor::module::wasmbus::host_error_len
    in wasmcloud_runtime::actor::module::call
    in wasmcloud_runtime::actor::module::call
    in wasmcloud_runtime::actor::module::handle
    in wasmcloud_host::wasmbus::handle_invocation with contract_id: "wasmcloud:httpserver", operation: "HttpServer.HandleRequest"
    in wasmcloud_host::wasmbus::handle_call
    in wasmcloud_host::wasmbus::handle_message

  TRACE wasmcloud_runtime::actor::module::wasmbus: wrote bytes into guest memory, buf: [102, 97, 105, 108, 101, 100, 32, 116, 111, 32, 99, 97, 108, 108, 32, 96, 119, 97, 115, 109, 99, 108, 111, 117, 100, 58, 98, 117, 115, 47, 104, 111, 115, 116, 46, 99, 97, 108, 108, 96, 58, 32, 102, 97, 105, 108, 101, 100, 32, 116, 111, 32, 99, 97, 108, 108, 32, 108, 105, 110, 107, 101, 100, 32, 112, 114, 111, 118, 105, 100, 101, 114, 58, 32, 102, 97, 105, 108, 101, 100, 32, 116, 111, 32, 112, 117, 98, 108, 105, 115, 104, 32, 111, 110, 32, 78, 65, 84, 83, 32, 116, 111, 112, 105, 99, 58, 32, 114, 101, 113, 117, 101, 115, 116, 32, 102, 97, 105, 108, 101, 100, 58, 32, 79, 116, 104, 101, 114]
    at crates/runtime/src/actor/module/wasmbus.rs:145
    in wasmcloud_runtime::actor::module::wasmbus::write_bytes with data: 1116160
    in wasmcloud_runtime::actor::module::wasmbus::write error into guest memory with err: "failed to call `wasmcloud:bus/host.call`: failed to call linked provider: failed to publish on NATS topic: request failed: Other"
    in wasmcloud_runtime::actor::module::wasmbus::host_error with err_ptr: 1116160
    in wasmcloud_runtime::actor::module::call
    in wasmcloud_runtime::actor::module::call
    in wasmcloud_runtime::actor::module::handle
    in wasmcloud_host::wasmbus::handle_invocation with contract_id: "wasmcloud:httpserver", operation: "HttpServer.HandleRequest"
    in wasmcloud_host::wasmbus::handle_call
    in wasmcloud_host::wasmbus::handle_message

  ERROR wasmcloud_host::wasmbus: failed to handle request, origin: WasmCloudEntity { public_key: "VAUO56KNNCEZAVKXM5K24QE72KAHCQ4RDTWMEOJX5JFG6IPQR3YJ2BXU", link_name: "httpserver", contract_id: "wasmcloud:httpserver" }, target: WasmCloudEntity { public_key: "MALHWDXB3I3C7YJXQJEYZ2FPVQB5CSWDLA7LKB6REPKGZYSPFJMYRYF3", link_name: "", contract_id: "" }, operation: "HttpServer.HandleRequest", invocation_id: "b6264478d0594093bb81df1abe983f10", e: failed to call actor: failed to call `__guest_call`: error while executing at wasm backtrace:
    0: 0xf9f6f - <unknown>!__rust_start_panic
    1: 0xf9ef4 - <unknown>!rust_panic
    2: 0xf9ee6 - <unknown>!std::panicking::rust_panic_with_hook::he017c20114c82dea
    3: 0xf956b - <unknown>!std::panicking::begin_panic_handler::{{closure}}::h5e23c6f065badef5
    4: 0xf9498 - <unknown>!std::sys_common::backtrace::__rust_end_short_backtrace::hff4b03aa315e3d86
    5: 0xf9b4d - <unknown>!rust_begin_unwind
    6: 0xfad72 - <unknown>!core::panicking::panic_fmt::h7859ddfa497c0e61
    7: 0x10145a - <unknown>!core::result::unwrap_failed::h99b8d970e94e9f11
    8: 0xa873d - <unknown>!core::result::Result<T,E>::expect::hdb4a9c60b082459f
    9: 0x77390 - <unknown>!<builtins_module_reactor::HttpLogRng as wasmcloud_actor::compat::http::Handler>::handle_request::h8e27fdc7bbcdd67c
   10: 0x4a8f0 - <unknown>!wasmcloud_actor::compat::http::<impl wasmcloud_actor::compat::Handler<dyn wasmcloud_actor::compat::http::Handler> for T>::handle::hb968e2410d84afd3
   11: 0x79166 - <unknown>!__guest_call
note: using the `WASMTIME_BACKTRACE_DETAILS=1` environment variable may show more debugging information: wasm trap: wasm `unreachable` instruction executed
    at crates/host/src/wasmbus/mod.rs:1339
    in wasmcloud_host::wasmbus::handle_message

  ERROR wasmcloud_host::wasmbus: failed to publish response to request, reply: "_INBOX.9SQ9hnpnsqlquGRZ99xbRA", e: PublishError(SendError { .. })
    at crates/host/src/wasmbus/mod.rs:1364
    in wasmcloud_host::wasmbus::handle_message

test wasmbus ... FAILED

failures:

---- wasmbus stdout ----
Error: component actor test failed

Caused by:
    0: failed to decode body as JSON
    1: EOF while parsing a value at line 1 column 0


failures:
    wasmbus

test result: FAILED. 0 passed; 1 failed; 0 ignored; 0 measured; 0 filtered out; finished in 24.45s

error: test failed, to rerun pass `--test wasmbus`
```

So if you search for `Keyvalue.Get`, you don't get it *anywhere* near the original operation error. With the original code you would have seen simply `Invocation failed`, but and the error, but no indication *which* invocation caused the error.

You get some indication in the targetting logs *before* the error, but after the `error!()` (which is before the actual invocation response with the error comes back) you don't get anything related to that.